tools/fixedpointarithmetic/src/main/java/dk/alexandra/fresco/decimal/fixed/BasicFixedNumeric.java

```diff
@@ -0,0 +1,266 @@
+package dk.alexandra.fresco.decimal.fixed;
+
+import dk.alexandra.fresco.decimal.BasicRealNumeric;
+import dk.alexandra.fresco.decimal.SReal;
+import dk.alexandra.fresco.decimal.fixed.utils.Truncate;
+import dk.alexandra.fresco.framework.DRes;
+import dk.alexandra.fresco.framework.builder.numeric.ProtocolBuilderNumeric;
+import dk.alexandra.fresco.framework.value.SInt;
+import java.math.BigDecimal;
+import java.math.BigInteger;
+import java.math.RoundingMode;
+
+public class BasicFixedNumeric implements BasicRealNumeric {
+
+  private final ProtocolBuilderNumeric builder;
+  private final BigInteger BASE = BigInteger.valueOf(2);
+  private final int defaultPrecision;
+  private final int maxPrecision;
+
+  public BasicFixedNumeric(ProtocolBuilderNumeric builder, int precision) {
+    this.builder = builder;
+    this.defaultPrecision = precision;
+
+    // We reserve as many bits the integer part as for the fractional part, and to be able to
+    // perform multiplications, we need to be able to represent at least two times that before
+    // truncation.
+    this.maxPrecision = builder.getBasicNumericContext().getMaxBitLength() / 4;
+    if (maxPrecision < 2 * defaultPrecision) {
+      throw new IllegalArgumentException(
+          "The precision was chosen too large for a product of two numbers to be representable "
+              + "in this numeric context. You cannot choose a precision larger than "
+              + (maxPrecision / 2) + ".");
+    }
+  }
+
+  public BasicFixedNumeric(ProtocolBuilderNumeric builder) {
```

Comment:
We should remember to document these constructors, in this case particularly the default precision used need to be known by the user of this class.